reverse_strand.py

```diff
@@ -0,0 +1,36 @@
+#! /usr/bin/python3
+import argparse
+from argparse import RawTextHelpFormatter
+import bamnostic as bs
+
+description="Reverses the allele for variants that got mapped onto the reverse strand of the new genome, and prints everything to a new file\nAlso filters out the non-mapped"
+
+parser = argparse.ArgumentParser(description = description, formatter_class=RawTextHelpFormatter)
+parser.add_argument("-i", "--bam", help="bam file containing remapped variants ")
+parser.add_argument("-o", "--outfile", help="name of new file")
+args = parser.parse_args()
+
+bamfile=bs.AlignmentFile(args.bam, 'rb')
+outfile=open(args.outfile, 'w')
+
+#reverses the allele for variants that got mapped onto the reverse strand of the new genome, and prints everything into correct columns
+
+for read in bamfile:
+	name=read.read_name
+	info=name.split("|")
+	#mapped onto reverse strand:
+	if read.flag & 16: #decoding bitwise flag with &; may need to be changed back to ==
+		if info[2]=='A':
+			rev_nucl="T"
+		elif info[2]=='T':
+			rev_nucl="A"
+		elif info[2]=='C':
+			rev_nucl="G"
+		elif info[2]=='G':
+			rev_nucl="C"
+		outfile.write("%s\t%s\t%s\t%s\t%s\t%s\t%s\n" % (read.reference_name,read.pos+51,info[3],rev_nucl,info[4],info[5],info[6]))
+	#only mapped onto direct strand:
+	elif read.flag==0:
+		outfile.write("%s\t%s\t%s\t%s\t%s\t%s\t%s\n" % (read.reference_name,read.pos+51,info[3],info[2],info[4],info[5],info[6]))
```

Comment:
This can be shortened by using https://biopython.org/wiki/Seq#complement-and-reverse-complement instead of the if-elif loop, and also only having one `outfile.write()` call. E. g.:

```python
nucl = info[2]
if read.flag & 16:
    nucl = Seq(nucl, generic_dna).complement()
outfile.write(...)
```

Reply:
I agree with the biopython addition. I have added it, along with the single `oufile.write`, although I do need an extra check before it so that I don't write the unmapped (flag=4) variants. (see new version when I push it)